src/Widget/Twig/TwigPreprocessor.php

```diff
@@ -460,19 +459,27 @@ public function preprocessEditorialLabel($cdbid, $settings)
             $publisher = $article['publisher'];
             $showPublisher = in_array($publisher, $settings['publishers']);
             $publisherInArray = in_array($publisher, $publishers);
-            if (!$publisherInArray && !$limitPublishers) {
-                $publishers[] = $publisher;
-            }
-            if (!$publisherInArray && ($limitPublishers && $showPublisher)) {
+            if (!$limitPublishers || $showPublisher) {
                 $publishers[] = $publisher;
             }
+
+            $publishers = array_unique($publishers);
```

Comment:
Best to put the `array_unique` outside of the `foreach` though. I don't think it would be a _terribly_ big performance impact, but it's only necessary once at the end of the loop anyway.